### ISSUE TITLE
Yoshi's Story missing bubbles and fruits.

### ISSUE THREAD

**Issue by retrobenny** [gonetz/GLideN64#1619]
I am using PJ64 and this happens.
The issue varies,as I also saw a parachute with an invisible melon,and now a room with many bubbles holding question items are invisible as seen in this snip...

![image](https://user-images.githubusercontent.com/16418716/31103175-4a4e8150-a79b-11e7-9112-9b75b3f67ba3.png)

I am on the Optimized Blending copy since I don't the more recent build which improves performance even more.
Depth Compare is not enabled currently,but that feature is mainly used in areas with water like Jungle Puddle.
I can post a save once I complete the level its in so it can be accessed directly on Trial Mode.
The name of it is Torrential Maze and its the 4th level of page 2,collecting all 3 happy hearts needed for manual access,do so in level 1 on page 1 for the easiest way to access it then stuff Yoshi's face full of fruit after finding all 3 hearts.

**Comment by Tasosgemah** [comment]
Can you use Depth compare for the whole game or does this cause issues in some areas?

**Comment by retrobenny** [comment]
I wouldn't mind if that could be an automatic setting for Yoshi's Story.

At least one other plugin renders the pause fruit icons and the bubbles and fruit nonstop so its GLideN64 with the occasional loss of stuff. (probably because of other plugins being hacks anyway)

I think Glide64 is the one that works for all of the basics like bubble rendering and all fruits showing up and the icons always being visible in the menu at all times.
Chalk this one up on the roadmap to Glide64 if its even there with Depth compare.
I will have to restart my save to recreate the scenario.

**Comment by retrobenny** [comment]
Platforms are invisible randomly as well. Have a video clip.

[2017-10-02 21-45-42.zip](https://github.com/gonetz/GLideN64/files/1350986/2017-10-02.21-45-42.zip)

Lost access because trying to open graphics randomly crashed,so I can't get an immediate image of the invisible platform,video makes more sense for this anyway.

**Comment by gonetz** [comment]
I tried to run on Torrential Maze, but I can't get invisible platforms or other issues. I use GLideN64 with default settings.

Are you sure that this kind of error is not emu core related? Please try to use other emulators and/or other versions of Project64.

**Comment by retrobenny** [comment]
After switching to the later version I was linked kinda recently,and running about in the stage a lot,I didn't notice the issue anymore,but that doesn't mean that Android is free of this issue,I can test it there sooner or later,hopefully real soon.

**Comment by retrobenny** [comment]
It definitely happens on Android as well,perfhaps worse than on Windows through PJ64.

![yoshi_story-006](https://user-images.githubusercontent.com/16418716/31179508-ca3274c0-a8e1-11e7-8396-023ed850d161.png)

I had to get 8 pictures in total when seeing things go missing.
The worst thing missing would be a random fruit that you needed or didn't know which kind it was if going for an all same fruit goal.

**Comment by gonetz** [comment]
Are you sure that it is graphics plugin problem? Please check with other plugins and with older versions of GLideN64.

**Comment by retrobenny** [comment]
That area renders fine normally but it just ended up that way in one instance.
Its behaving in mostly the exact same way as it did on Windows.
I could try the older GLideN64 version/s via Windows but it would be a big pain to revert to older versions on Android.

**Comment by retrobenny** [comment]
On Public Release 2.0 there is this odd square that I also happened to run into on Android with the latest updates.

![image](https://user-images.githubusercontent.com/16418716/31197169-db7eaaee-a915-11e7-8989-7069e4c698d8.png)

**Comment by retrobenny** [comment]
It gets worse with PR2.

![image](https://user-images.githubusercontent.com/16418716/31201269-be709c88-a923-11e7-919d-fd5157c8e68d.png)

Yoshi in the sky with numbers. Even Yoshi is confused.

**Comment by fzurita** [comment]
It sounds like some sort of alpha issue. Do the missing elements have any special effects added to them?

**Comment by gonetz** [comment]
Yes, it can be another issue with blending. Have you tried to reproduce that bug with legacy blending enabled?

**Comment by retrobenny** [comment]
[YOSHI STORY.zip](https://github.com/gonetz/GLideN64/files/1357925/YOSHI.STORY.zip)

Here is my save with every area hastily unlocked normally but with survival cheats like action swaps and Super Happy forced enabled to save time.

**Comment by gonetz** [comment]
Do you know fast scenario to reproduce the problem?

**Comment by retrobenny** [comment]
The level called Shy Guy Limbo is the one with that picture,and there is a climbing ball near a blue pipe in the level with said pipe taking you to the exact problem area.

You can use this code to navigate to it swiftly,but maybe the delay brings on the issue.

Jump Swims In Air
8002DB3F 004E

Or just this code to make things easier with invincibility...

Always Have Super Happy Powers
800F8946 0009
Always have eggs,invincible to enemies,longer reach,better fluttering,etc.

Otherwise it needs a keen eye or the proper maneuver with Lakitu's cloud to reach the pipe in a fairly reasonable amount of time.

**Comment by daninthemix** [comment]
Literally the final boss is invisible.

**Comment by olivieryuyu** [comment]
can u provide a savestate of the final boss?

**Comment by purplemarshmallow** [comment]
Is it a framebuffer emulation bug maybe?
What happens if copy auxiliary buffers is enabled?
What happens is framebuffer emulation is disabled?
Does it bypass the issue?

**Comment by IronYoshi64** [comment]
I know this is an old post, but I think I figured out how to completely resolve this issue. In the GLideN64.custom.ini file, have this for Yoshi's Story:

[YOSHI%20STORY]
Good_Name=Yoshi's Story (E)(U)
frameBufferEmulation\nativeResFactor=0
generalEmulation\enableShadersStorage=0
generalEmulation\correctTexrectCoords=0
generalEmulation\enableNativeResTexrects=1

I have done a lot of testing with these settings, and I have yet to have it happen with these. I recommend you delete the .shaders and .keys for the game inside the shaders folder to be sure.
